sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/Statsbeat.cs

```diff
@@ -38,8 +51,86 @@ static Statsbeat()
 
         private static Measurement<int> GetAttachStatsBeat()
         {
-            // TODO: Add additional properties required for statbeat
-            return new(1, new("cikey", Customer_Ikey), new("language", "dotnet"));
+            if (s_resourceProvider == null)
+            {
+                SetResourceProviderDetails();
+            }
+            else if (s_resourceProviderId == "vm")
+            {
+                // Query again as Id might have been updated.
```

Comment:
If you are thinking about vmId it should only get updated on restart.

Reply:
removed this.